modules/auxiliary/gather/ssllabs_scan.rb

```diff
@@ -142,18 +156,30 @@ def self.has_object_ref(name, klass)
 
     def self.load(attributes = {})
       obj = self.new
+      unused_attrs = []
       attributes.each do |name, value|
         if @fields.include?(name)
           obj.instance_variable_set("@#{name}", value)
         elsif @lists.key?(name)
-          obj.instance_variable_set("@#{name}", value.map { |v| @lists[name].load(v) }) unless value.nil?
+          unless value.nil?
+            var = value.map do |v|
+              val, ua = @lists[name].load(v)
+              unused_attrs.concat ua
+              val
+            end
+            obj.instance_variable_set("@#{name}", var)
```

Comment:
How safe is this on externally supplied data?

Reply:
Note that this is just an unwrapped version of the original one-liner. I don't see a risk given the amount of validation going on here. Are you thinking of a deserialization attack?